configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/screening/ScreenerFragment.kt

```diff
@@ -0,0 +1,160 @@
+    super.onCreateOptionsMenu(menu, inflater)
+    inflater.inflate(R.menu.screen_encounter_fragment_menu, menu)
+  }
+
+  override fun onOptionsItemSelected(item: MenuItem): Boolean {
+    return when (item.itemId) {
+      R.id.action_add_patient_submit -> {
+        onSubmitAction()
+        true
+      }
+      android.R.id.home -> {
+        showCancelScreenerQuestionnaireAlertDialog()
+        true
+      }
+      else -> super.onOptionsItemSelected(item)
+    }
+  }
+
+  private fun setUpActionBar() {
+    (requireActivity() as AppCompatActivity).supportActionBar?.apply {
+      setDisplayHomeAsUpEnabled(true)
+    }
+  }
+
+  private fun setViewModelQuestionnaire() {
+    viewModel.questionnaireString = args.questionnaireString
+  }
+
+  private fun addQuestionnaireFragment() {
+    childFragmentManager.commit {
+      add(
+        R.id.add_patient_container,
+        QuestionnaireFragment.builder().setQuestionnaire(viewModel.questionnaireString).build(),
+        QUESTIONNAIRE_FRAGMENT_TAG
+      )
+    }
+  }
+
+  private fun onSubmitAction() {
+    val questionnaireFragment =
+      childFragmentManager.findFragmentByTag(QUESTIONNAIRE_FRAGMENT_TAG) as QuestionnaireFragment
+    viewModel.saveScreenerEncounter(
+      questionnaireFragment.getQuestionnaireResponse(),
+      args.patientId
+    )
+  }
+
+  private fun showCancelScreenerQuestionnaireAlertDialog() {
+    val alertDialog: AlertDialog? =
+      activity?.let {
+        val builder = AlertDialog.Builder(it)
+        builder.apply {
+          setMessage(getString(R.string.cancel_questionnaire_message))
+          setPositiveButton(getString(android.R.string.yes)) { _, _ ->
+            NavHostFragment.findNavController(this@ScreenerFragment).navigateUp()
+          }
+          setNegativeButton(getString(android.R.string.no)) { _, _ -> }
+        }
+        builder.create()
+      }
+    alertDialog?.show()
+  }
+
+  private fun onBackPressed() {
+    activity?.onBackPressedDispatcher?.addCallback(viewLifecycleOwner) {
+      showCancelScreenerQuestionnaireAlertDialog()
+    }
+  }
+
+  private fun observeResourcesSaveAction() {
+    viewModel.screenerState.observe(viewLifecycleOwner) {
+      if (!it.isResourceSaved) {
+        Snackbar.make(
+            requireActivity().findViewById(android.R.id.content),
+            R.string.inputs_missing,
+            Snackbar.LENGTH_SHORT
+          )
+          .show()
+        return@observe
+      }
+      Snackbar.make(
+          requireActivity().findViewById(android.R.id.content),
+          R.string.resources_saved,
+          Snackbar.LENGTH_SHORT
+        )
+        .show()
+      careWorkflowExecutionViewModel.updateTaskAndCarePlanStatus(
+        args.taskLogicalId,
+        Task.TaskStatus.COMPLETED,
+        it.encountersCreated,
+        true,
+      )
+      NavHostFragment.findNavController(this).navigateUp()
+    }
+  }
+
+  companion object {
```

Comment:
Are `EXTRA_QUESTIONNAIRE_JSON_STRING`, `CURRENT_TASK_ID` and `MODIFIED_TASK_ID` being used anywhere?

Reply:
Nope. Removed.